src/analyses/constant_propagator.cpp

```diff
@@ -159,19 +95,17 @@ void constant_propagator_domaint::transform(
     exprt g;
 
     if(from->get_target()==to)
-      g = simplify_expr(from->guard, ns);
+      g=simplify_expr(from->guard, ns);
     else
-      g = simplify_expr(not_exprt(from->guard), ns);
+      g=simplify_expr(not_exprt(from->guard), ns);
 
-    if (g.is_false())
+    if(g.is_false())
      values.set_to_bottom();
     else
     {
-      //TODO: we need to support widening!
-      if (g.is_constant())
-        values.set_to_top();
-      else
-        two_way_propagate_rec(g, ns);
+      two_way_propagate_rec(g, ns);
+      INVARIANT(!values.is_bottom,
```

Comment:
// If two way propagate is enabled then it may be possible to detect
      // that the branch condition is infeasible and thus the domain should
      // be set to bottom.  Without it the domain will only be set to bottom
      // if the guard expression is trivially (i.e. without context) false.

Reply:
Also added to the code...